x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_logic.ts

```diff
@@ -155,4 +287,270 @@ export const RelevanceTuningLogic = kea<
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+    },
+    onSearchSettingsSuccess: ({ searchSettings }) => {
+      actions.setSearchSettingsResponse(searchSettings);
+      actions.getSearchResults();
+      window.scrollTo(0, 0);
+    },
+    onSearchSettingsError: () => {
+      window.scrollTo(0, 0);
+    },
+    updateSearchSettings: async () => {
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      const url = `/api/app_search/engines/${engineName}/search_settings`;
+
+      try {
+        const response = await http.put(url, {
+          body: JSON.stringify(removeBoostStateProps(values.searchSettings)),
+        });
+        setSuccessMessage(UPDATE_SUCCESS_MESSAGE);
+        actions.onSearchSettingsSuccess(response);
+      } catch (e) {
+        flashAPIErrors(e);
+        actions.onSearchSettingsError();
+      }
+    },
+    resetSearchSettings: async () => {
+      if (window.confirm(RESET_CONFIRMATION_MESSAGE)) {
+        const { http } = HttpLogic.values;
+        const { engineName } = EngineLogic.values;
+
+        const url = `/api/app_search/engines/${engineName}/search_settings/reset`;
+
+        try {
+          const response = await http.post(url);
+          setSuccessMessage(DELETE_SUCCESS_MESSAGE);
+          actions.onSearchSettingsSuccess(response);
+        } catch (e) {
+          flashAPIErrors(e);
+          actions.onSearchSettingsError();
+        }
+      }
+    },
+    toggleSearchField: ({ name, disableField }) => {
+      const { searchSettings } = values;
+      const { search_fields: searchFields } = searchSettings;
+
+      actions.setSearchSettings({
+        ...searchSettings,
+        boosts: searchSettings.boosts,
+        search_fields: {
+          ...omit(searchFields, name),
+          ...(!disableField && { [name]: { weight: 1 } }),
+        },
+      });
+      actions.getSearchResults();
+    },
+    updateFieldWeight: ({ name, weight }) => {
+      const { searchSettings } = values;
+      const { search_fields: searchFields } = searchSettings;
+
+      actions.setSearchSettings({
+        ...searchSettings,
+        boosts: searchSettings.boosts,
+        search_fields: {
+          ...searchFields,
+          [name]: {
+            ...searchFields[name],
+            weight: Math.round(weight * 10) / 10,
+          },
+        },
+      });
+      actions.getSearchResults();
+    },
+    addBoost: ({ name, type }) => {
+      const { searchSettings } = values;
+      const { boosts } = searchSettings;
+      const emptyBoost = { type, factor: 1, newBoost: true };
+      let boostArray;
+
+      if (Array.isArray(boosts[name])) {
+        boostArray = boosts[name].slice();
+        boostArray.push(emptyBoost);
+      } else {
+        boostArray = [emptyBoost];
+      }
+
+      actions.setSearchSettings({
+        ...searchSettings,
+        boosts: {
+          ...boosts,
+          [name]: boostArray,
+        },
+      });
```

Comment:
Just wondering, why do we not call `actions.getSearchResults();` at the end for this one listener? We seem to be doing so for every other listener that calls `actions.setSearchSettings()`

Reply:
Also, one thought about DRYing this out a little bit so we don't have to manually call `actions.getSearchResults()` every time - can we add a new listener called `setSearchSettings`, and do this:

```js
listeners: ({ actions, values }) => ({
  setSearchSettings: () => {
    actions.getSearchResults();
  },
```

I _think_ this should add an extra listener after the existing `setSearchSettings` action which automatically calls getSearchResults

Reply:
I noticed that as well, and I wasn't sure whether that was intentional or not.

Digging into it a bit more, I *think* maybe it's because when you "add" a boost, you're actually adding an empty boost that does not yet have a value and wouldn't have any effect on results.

However, it would also be harmless to run the query an extra time. I'll make your first suggested change, which should clean this up quite a bit. Good catch.

Reply:
https://github.com/elastic/kibana/pull/89461/commits/b0a1ec25f1187f1cb843b3817b0a86066b93816f

Reply:
Nice, I love it!